[Code Review - Do Not Merge]

idcmp opened the pull request.

idcmp committed:
-  add led support

idcmp committed:
- cleanup

daschuer commented on build/depends.py:
```diff
@@ -1236,6 +1236,8 @@ def sources(self, build):
                    "util/indexrange.cpp",
                    "util/widgetrendertimer.cpp",
 
+                   "idcmp/junction.cpp",
```
You should find a reasonable folder name for your feature, related to the feature itself.

daschuer commented on src/idcmp/junction.cpp:
```diff
@@ -0,0 +1,55 @@
+        : QObject(),
+          m_pPlayerManager(pPlayerManager) {
+
+    for (unsigned int deckNum = 1; deckNum <= pPlayerManager->numberOfDecks(); deckNum++) {
```
We have also the [Master] num_decks you can use.

daschuer reviewed (changes requested):
Nice! Do you have an idea how to make it usable for all users?

daschuer commented on src/mixxx.h:
```diff
@@ -165,6 +167,8 @@ class MixxxMainWindow : public QMainWindow {
 #endif
     ControllerManager* m_pControllerManager;
 
+    IdcmpJunction *m_pIdcmpJunction;
```
We alligne the * with the type.

daschuer commented on src/idcmp/junction.cpp:
```diff
@@ -0,0 +1,55 @@
+
+    file.seek(12);
+    // thinklight off or thinklight blink
+    file.putChar((v == 0) ? 0x0a : 0xca);
```
Here you should use the chapter litheral
'\r' what is 0xca?

idcmp replied:
Thanks for taking the time to review this!  

It's actually two nibbles. The first specifies either on/off/blink, the second specifies which light. More details here: https://www.reddit.com/r/thinkpad/comments/7n8eyu/thinkpad_led_control_under_gnulinux/

idcmp committed:
- make the end of track colour cyan
- don't show crates starting with "~"
- remove option to destroy the entire autodj queue

Holzhaus commented:
I have not the slightest idea what this does. Close?

uklotzde commented:
@idcmp Please open a new PR with a meaningful title and description when ready. Closing.
